src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,195 @@
+import { buildHierarchy } from './buildHierarchy.mjs';
+import { getRemarkRehype } from '../../../utils/remark.mjs';
+import { transformNodesToString } from '../../../utils/unist.mjs';
+import { parseList } from './parseList.mjs';
+
+const sectionTypePlurals = {
+  module: 'modules',
+  misc: 'miscs',
+  class: 'classes',
+  method: 'methods',
+  property: 'properties',
+  global: 'globals',
+  example: 'examples',
+  ctor: 'signatures',
+  classMethod: 'classMethods',
+  event: 'events',
+  var: 'vars',
+};
+
+/**
+ * Converts a value to an array.
+ * @template T
+ * @param {T | T[]} val - The value to convert.
+ * @returns {T[]} The value as an array.
+ */
+const enforceArray = val => (Array.isArray(val) ? val : [val]);
+
+/**
+ * Creates metadata from a hierarchized entry.
+ * @param {import('../types.d.ts').HierarchizedEntry} entry - The entry to create metadata from.
+ * @returns {import('../types.d.ts').Meta} The created metadata.
+ */
+function createMeta(entry) {
+  const {
+    added_in = [],
+    n_api_version = [],
+    deprecated_in = [],
+    removed_in = [],
+    changes,
+  } = entry;
+
+  return {
+    changes,
+    added: enforceArray(added_in),
+    napiVersion: enforceArray(n_api_version),
+    deprecated: enforceArray(deprecated_in),
+    removed: enforceArray(removed_in),
+  };
+}
+
+/**
+ * Creates a section from an entry and its heading.
+ * @param {import('../types.d.ts').HierarchizedEntry} entry - The AST entry.
+ * @param {HeadingMetadataParent} head - The head node of the entry.
+ * @returns {import('../types.d.ts').Section} The created section.
+ */
+function createSection(entry, head) {
+  return {
+    textRaw: transformNodesToString(head.children),
+    name: head.data.name,
+    type: head.data.type,
+    meta: createMeta(entry),
+    introduced_in: entry.introduced_in,
+  };
+}
+
+/**
+ * Parses stability metadata and adds it to the section.
+ * @param {import('../types.d.ts').Section} section - The section to add stability to.
+ * @param {Array} nodes - The AST nodes.
+ * @param {import('../types.d.ts').HierarchizedEntry} entry - The entry to handle.
+ */
+function parseStability(section, nodes, entry) {
+  const json = entry.stability.toJSON()[0];
+  if (json) {
+    section.stability = json.index;
+    section.stabilityText = json.description;
+    nodes.splice(0, 1);
+  }
+}
+
+let lazyHTML;
```

Comment:
Is this a top-level let? This is an issue, we shouldn't store top-level variables for what is run-specific. This possibly works with a CLI, but if this is used elsewhere, we will never erase it. 

I believe you can do something similar from our queries.mjs file which creates an instance of the query builder, here you create an instance of the section builder.